scraper/src/youtube2zim/scraper.py

```diff
@@ -1108,3 +1120,240 @@ def to_data_js(video):
+                    raise
+                return Subtitle(
+                    code=lang,
+                    name=f"{subtitle['english'].title()} - {subtitle['query']}",
+                )
+
+            # Youtube.com sorts subtitles by English name
+            return sorted(map(to_subtitle_object, languages), key=lambda x: x.name)
+
+        def get_videos_list(playlist):
+            videos = load_mandatory_json(
+                self.cache_dir, f"playlist_{playlist.playlist_id}_videos"
+            )
+            videos = list(filter(skip_deleted_videos, videos))
+            videos = list(filter(is_present, videos))
+            videos = list(filter(has_channel, videos))
+            videos = sorted(videos, key=lambda v: v["snippet"]["position"])
+            return videos
+
+        def generate_video_object(video):
+            video_id = video["contentDetails"]["videoId"]
+            title = video["snippet"]["title"]
+            description = video["snippet"]["description"]
+            publication_date = video["contentDetails"]["videoPublishedAt"]
+            author = videos_channels[video_id]
+
+            return Video(
+                id=video_id,
+                title=title,
+                description=description,
+                author=Author(
+                    channel_id=author["channelId"],
+                    channel_title=author["channelTitle"],
+                    profile_path=f"channels/{author['channelId']}/profile.jpg",
+                    banner_path=f"channels/{author['channelId']}/banner.jpg",
+                ),
+                publication_date=publication_date,
+                video_path=f"videos/{video_id}/video.{self.video_format}",
+                thumbnail_path=get_thumbnail_path(video_id),
+                subtitle_path=f"videos/{video_id}",
+                subtitle_list=get_subtitles(video_id),
+                duration=0,
+            )
+
+        def generate_video_preview_object(video):
+            video_id = video["contentDetails"]["videoId"]
+            title = video["snippet"]["title"]
+            slug = get_video_slug(video)
+            return VideoPreview(
+                slug=slug,
+                id=video_id,
+                title=title,
+                thumbnail_path=get_thumbnail_path(video_id),
+                duration=0,
+            )
+
+        def get_video_slug(video):
+            title = video["snippet"]["title"]
+            video_id = video["contentDetails"]["videoId"]
+            return f"{get_slug(title)}-{video_id[:4]}"
+
+        def generate_playlist_object(playlist):
+            videos = get_videos_list(playlist)
+            videos_list = []
+            for video in videos:
+                videos_list.append(generate_video_preview_object(video))
+            return Playlist(
+                id=playlist.playlist_id,
+                title=playlist.title,
+                description=playlist.description,
+                videos=videos_list,
```

Comment:
I would rewrite it with something like (not tested) `videos = [generate_video_preview_object(video) for video in get_videos_list(playlist)]` (this one-liner get is both easier to read and more concise)

Reply:
Refactored in [8ff4342](https://github.com/openzim/youtube/pull/219/commits/8ff43429631f402e4e51d66853f9aa2f88ef8d7f)